### PR TITLE
move interfaces/ into workflow/; small import fixups elsewhere

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
All execution code should ultimately be in the same place, and the existing `interfaces` sub-package should be _part_ of that place. However, I do not want to break existing paths/imports until any redesign is complete.

**Description of the Change:**
Moves `pennylane/interfaces/` to `pennylane/workflow/interfaces/`, but preserves the old import naming (with the additional alias of `pennylane.workflow.interfaces` existing, touched up in a few places to ensure import order in the top level initfile is still intact).

**Benefits:**
We will have a new home for all sorts of execution pipeline code, and it's clearer what the interfaces package exists for.

**Possible Drawbacks:**
N/A

### CODE REVIEW

**Comment by timmysilv** [comment]
[qml_interfaces docs](https://xanaduai-pennylane--3634.com.readthedocs.build/en/3634/code/qml_interfaces.html) appear to be alright! One catch - aside from `execute`, they now all show the name `pennylane.workflow.interfaces.*` as their titles

**Review by albi3ro — COMMENTED** [review]
In my mind, I think it would make sense to move `execution.py` and `set_shots.py` into the `workflow` folder itself and out of the `interfaces` submodule. Though `INTERFACE_MAP` and `SUPPORTED_INTERFACES` should still be defined in the interfaces module.

**Comment by timmysilv** [comment]
I agree, I just didn't want to move anything around in this PR. If you think there's nothing to gain from waiting, I'll do it here

File: pennylane/workflow/__init__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
```
Comment (rmoyard) [inline pennylane/workflow/__init__.py:1]:
```suggestion
# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
```

**Review by rmoyard — APPROVED** [review]
It looks good to me, thank you @timmysilv ! I would say better to move the execute in a separate PR.

**Comment by timmysilv** [comment]
@rmoyard I'm doing it all in one PR because that's effectively the point of this PR. I thought about splitting it up but I feel like that next PR would almost look like I just forgot to do it here 😋 

You can still use `qml.execute` directly, but I should note that pretty much everything else has to go through `workflow.interfaces` now. I've tried to set up some aliasing, but whatever I do leads to some problem or another:
- adding `pennylane/interfaces.py` complains that `pennylane.interfaces` isn't a module, especially when trying to import from files in the folder (like `jax.py`) rather than top-level imported methods and stuff
- adding `pennylane/interfaces/__init__.py` is similar
- symlinking `pennylane/interfaces` to `pennylane/workflow/interfaces` works well (one catch for namespacing of the error but it's forgivable), but only locally because it needs a global symlink which isn't supported for python packages that are released publicly

tl;dr this is a breaking change, and we should proceed with caution.

**Comment by timmysilv** [comment]
probably won't do this at all
